contracts/root/withdrawManager/WithdrawManagerProxy.sol

```diff
@@ -1,14 +1,16 @@
 pragma solidity ^0.5.2;
 
-import { Registry } from '../Registry.sol';
-import { Proxy } from '../../common/misc/Proxy.sol';
-import { WithdrawManagerStorage } from './WithdrawManagerStorage.sol';
+import { Registry } from "../../common/Registry.sol";
+import { Proxy } from "../../common/misc/Proxy.sol";
+import { WithdrawManagerStorage } from "./WithdrawManagerStorage.sol";
+import { RootChain } from "../RootChain.sol";
 
 contract WithdrawManagerProxy is Proxy, WithdrawManagerStorage {
-  constructor(address _proxyTo, address _registry)
+  constructor(address _proxyTo, address _registry, address _rootChain)
```

Comment:
For now, I assumed that `rootChain` will be deployed first and that can be passed to constructor to other contracts. To avoid cyclic deps, if rootChain needs to access other contracts, it'll read from registry - so basically, only the registry will have a setter. Thoughts?

Reply:
how we are planning to update the local copy of rootChain then?

Reply:
Upgreadable via the proxy pattern.